code/items/base/Ammo.cs

```diff
@@ -72,10 +79,18 @@ private void GiveAmmo( Player player )
 		if ( !player.Inventory.HasWeaponOfAmmoType( Type ) )
 			return;
 
-		player.GiveAmmo( Type, CurrentCount );
+		int iMaxAmmo = Inventory.GetAmmoLimit( Type );
+		int iAdd = Math.Min( CurrentCount, iMaxAmmo - player.AmmoCount( Type ) );
+		if ( iAdd <= 0 ) return;
```

Comment:
Don't touch this system. Just add a check if we can add the ammo because of the ammo cap. We want the ammo boxes to behave like the ones in HL2.